### PR TITLE
[SPARK-21323][SQL]Rename plans.logical.statsEstimation.Range to ValueInterval

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

Rename org.apache.spark.sql.catalyst.plans.logical.statsEstimation.Range to ValueInterval.
The current naming is identical to logical operator "range". 
Refactoring it to ValueInterval is more accurate.

## How was this patch tested?

unit test

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @rxin @gatorsmile

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/ValueInterval.scala [hunk 1]
```diff
@@ -38,46 +38,49 @@ case class NumericRange(min: Decimal, max: Decimal) extends Range {
...
+      NumericValueInterval(
         min = EstimationUtils.toDecimal(min.get, dataType),
         max = EstimationUtils.toDecimal(max.get, dataType))
   }
 
-  def isIntersected(r1: Range, r2: Range): Boolean = (r1, r2) match {
-    case (_, _: DefaultRange) | (_: DefaultRange, _) =>
+  def isIntersected(r1: ValueInterval, r2: ValueInterval): Boolean = (r1, r2) match {
+    case (_, _: DefaultValueInterval) | (_: DefaultValueInterval, _) =>
       // The DefaultRange represents string/binary types which do not have max/min stats,
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/ValueInterval.scala]: `DefaultRange` -> `DefaultValueInterval `

**Comment by gatorsmile** [comment]
LGTM except one minor comment.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/ValueInterval.scala (hunk 1 above)
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/ValueInterval.scala]: Thanks, I have fixed it

**Comment by gengliangwang** [comment]
retest this please

**Comment by rxin** [comment]
Merging in master.

**Pull request closed without merging**